soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -1191,81 +1191,537 @@ s16 Randomizer::GetRandomizedItemId(GetItemID ogId, s16 actorId, s16 actorParams
 }
 
 s16 Randomizer::GetItemFromActor(s16 actorId, s16 actorParams, s16 sceneNum, GetItemID ogItemId) {
-    return GetItemFromGet(this->itemLocations[GetCheckFromActor(sceneNum, actorId, actorParams)], ogItemId);
+    return GetItemIDFromRandomizerGet(this->itemLocations[GetCheckFromActor(sceneNum, actorId, actorParams)], ogItemId);
 }
 
-s16 Randomizer::GetItemFromGet(RandomizerGet randoGet, GetItemID ogItemId) {
+CanObtainResult Randomizer::GetCanObtainFromRandomizerCheck(RandomizerCheck randomizerCheck) {
```

Comment:
might be worth splitting this off into a `CanObtainRandomizerGet` function that takes in the RG, and then calling that from this function (not sure why exactly we'd need that functionality but i feel like it'd be nice to keep each function on topic a bit better

Reply:
It was like this previously, but I had to move the call up to the global level for other reasons, and randomizer class currently doesn't expose a method to get a RandomizerGet value. I will potentially address this in the follow up renaming things PR

Reply:
i see that you added it to the header in https://github.com/HarbourMasters/Shipwright/pull/1379, but in `randomizer.cpp` i still see
```cpp
ItemObtainability Randomizer::GetItemObtainabilityFromRandomizerCheck(RandomizerCheck randomizerCheck) {
    switch (this->itemLocations[randomizerCheck]) {
        case RG_NONE:
```
it seems the work had been done in https://github.com/HarbourMasters/Shipwright/pull/1379/commits/5961c33ba9ac80c152ea4f266500f6ace43bcc7a, maybe something got list in a merge?

Reply:
Yea I commented down below regarding this

> As of https://github.com/HarbourMasters/Shipwright/pull/1379 I am able to do this if desired, but I don't feel it alleviates my uncomfortable-ness here

Reply:
ah, even if we aren't using it there, i still think we should split up the logic so we have `GetItemObtainabilityFromRandomizerCheck` call into `GetItemObtainabilityFromRandomizerGet`

only having that functionality available at the RC level instead of at the RG level when the first line of the function is getting the RG from the RC just feels kind of odd to me

Reply:
Oh, actually I was confused and I see what you're saying now, it did get lost somehow, pushed up another commit to that PR